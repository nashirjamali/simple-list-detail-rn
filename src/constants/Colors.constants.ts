export const COLORS = {
  BRAND: '#FB6544',
  BLACK: '#121212',
  SUCCESS: '#55B585',
  GRAY: '#999999',
  WHITE: '#FFFFFF',
} as const;
