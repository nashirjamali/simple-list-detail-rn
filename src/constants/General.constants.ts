export const API_URL =
  'https://recruitment-test.flip.id/frontend-test' as const;
