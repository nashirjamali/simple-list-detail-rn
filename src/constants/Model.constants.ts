export const TRANSACTION_STATUS = {
  SUCCESS: 'SUCCESS',
  PENDING: 'PENDING',
} as const;
