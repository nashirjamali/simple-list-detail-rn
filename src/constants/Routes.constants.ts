export const ROUTES = {
  TRANSACTION_LIST: 'transactionsList',
  TRANSACTION_DETAILS: 'transactionsDetails',
} as const;
