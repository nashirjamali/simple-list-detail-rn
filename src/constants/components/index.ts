export * from './GeneralText.constants';
export * from './Button.constants';
export * from './Label.constants';
export * from './Card.constants';
