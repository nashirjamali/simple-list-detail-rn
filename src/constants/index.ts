export * from './components';
export * from './Colors.constants';
