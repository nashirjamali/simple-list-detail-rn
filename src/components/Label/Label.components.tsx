import React, {FC, ReactElement} from 'react';
import {View} from 'react-native';

import type {LabelProps} from '../../@types';
import GeneralText from '../GeneralText/GeneralText.components';
import {COLORS, GENERAL_TEXT_VARIANTS, LABEL_VARIANTS} from '../../constants';
import styles from './Label.styles';

/**
 * Renders a label with a pending style.
 *
 * @param {string} text - The text to be displayed in the label.
 * @returns {ReactElement} The rendered label component for the pending state.
 */
const _renderPending = (text: string): ReactElement => (
  <View style={styles.containerPending}>
    <GeneralText
      variant={GENERAL_TEXT_VARIANTS.NORMAL}
      style={{textAlign: 'center', fontWeight: 'bold'}}>
      {text}
    </GeneralText>
  </View>
);

/**
 * Renders a label with a success style.
 *
 * @param {string} text - The text to be displayed in the label.
 * @returns {ReactElement} The rendered label component for the success state.
 */
const _renderSuccess = (text: string): ReactElement => (
  <View style={styles.containerSuccess}>
    <GeneralText
      variant={GENERAL_TEXT_VARIANTS.NORMAL}
      style={{textAlign: 'center', fontWeight: 'bold', color: COLORS.WHITE}}>
      {text}
    </GeneralText>
  </View>
);

/**
 * Maps the variant to the corresponding label rendering function.
 *
 * @param {LabelProps} props - The properties for the label component.
 * @param {string} props.text - The text to be displayed in the label.
 * @param {LABEL_VARIANTS} props.variant - The variant type for styling the label.
 * @returns {ReactElement} The rendered label component based on the variant.
 */
const _mappingVariant = ({text, variant}: LabelProps): ReactElement => {
  switch (variant) {
    case LABEL_VARIANTS.PENDING:
      return _renderPending(text);

    case LABEL_VARIANTS.SUCCESS:
      return _renderSuccess(text);

    default:
      return <></>; // Return an empty fragment for unsupported variants
  }
};

/**
 * A label component that displays text with different styles based on the variant.
 *
 * @param {LabelProps} props - The properties for the Label component.
 * @param {string} props.text - The text to be displayed in the label.
 * @param {LABEL_VARIANTS} props.variant - The variant type for styling the label.
 * @returns {JSX.Element} The rendered label component.
 */
const Label: FC<LabelProps> = (props: LabelProps): JSX.Element => (
  <View>{_mappingVariant(props)}</View>
);

export default Label;
