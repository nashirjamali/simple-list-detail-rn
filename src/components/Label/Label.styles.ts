import {StyleSheet} from 'react-native';

import type {LabelStyles} from '../../@types';

import {COLORS} from '../../constants';

const styles: LabelStyles = StyleSheet.create({
  containerOnCheck: {
    borderRadius: 10,
    padding: 10,
    borderWidth: 2,
    borderColor: COLORS.BRAND,
    justifyContent: 'center',
  },
  containerSuccess: {
    borderRadius: 10,
    padding: 10,
    justifyContent: 'center',
    backgroundColor: COLORS.SUCCESS,
  },
});

export default styles;
