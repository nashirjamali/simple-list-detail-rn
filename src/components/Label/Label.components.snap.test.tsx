import React from 'react';
import {render} from '@testing-library/react-native';

import {LABEL_VARIANTS} from '../../constants';

import Label from './Label.components';

describe('Label Component', () => {
  it('renders correctly with pending variant', () => {
    const {toJSON} = render(
      <Label text="Pending..." variant={LABEL_VARIANTS.PENDING} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with success variant', () => {
    const {toJSON} = render(
      <Label text="Success!" variant={LABEL_VARIANTS.SUCCESS} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
