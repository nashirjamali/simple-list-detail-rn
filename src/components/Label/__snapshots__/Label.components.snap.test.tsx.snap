// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component renders correctly with pending variant 1`] = `
<View>
  <View
    style={
      {
        "borderColor": "#FB6544",
        "borderRadius": 10,
        "borderWidth": 2,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#121212",
          "fontSize": 14,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Pending...
    </Text>
  </View>
</View>
`;

exports[`Label Component renders correctly with success variant 1`] = `
<View>
  <View
    style={
      {
        "backgroundColor": "#55B585",
        "borderRadius": 10,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 14,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Success!
    </Text>
  </View>
</View>
`;
